lib/ferto/client.rb

```diff
@@ -77,18 +80,25 @@ def download(aggr_id:, aggr_limit: @aggr_limit, url:,
     def build_header(aggr_id)
       {
         'Content-Type': 'application/json',
-        'X-Aggr':      aggr_id.to_s
+        'X-Aggr': aggr_id.to_s
       }
     end
 
-    def build_body(aggr_id, aggr_limit, url, callback_url, mime_type, extra)
+    def build_body(aggr_id, aggr_limit, url, callback_url, callback_type,
+                   callback_dst, mime_type, extra)
       body = {
         aggr_id: aggr_id,
         aggr_limit: aggr_limit,
-        url: url,
-        callback_url: callback_url
+        url: url
       }
 
+      if !callback_url.empty?
+        body[:callback_url] = callback_url
+      else
+        body[:callback_type] = callback_type
+        body[:callback_dst] = callback_dst
+      end
+
```

Comment:
What if both `callback_url` and `callback_dst` are empty?

Reply:
This is handled in the downloader. We will get a `BadRequest`

Reply:
True. And since this is temporary (callback_url will go away soon), this is fine.